src/v/storage/disk_log_impl.cc

```diff
@@ -435,8 +435,38 @@ ss::future<> disk_log_impl::compact_adjacent_segments(
           segments.back(), "compact_adjacent_segments");
     }
 }
+compaction_config
+disk_log_impl::get_final_comapction_config(compaction_config defaults) const {
```

Comment:
why `get_final*`  are there 2 methods  in `disk_log_impl`?

Reply:
Configuration is passed in from `storage::log_manager` some of the properties may be overridden by log specific settings hence the name. I can rename if you find this misleading.

Reply:
i was asking for what `final` means. let's rename to `get_compaction_config() const` i feel like there should be a free standing method that does the merging of `defaults` 

```
      config merge_defaults(log, defaults) {
         auto ret = log->get_compaction_config()
         // override with defaults
        return ret;
      }
      auto x = merge_defaults(log, defaults)
```

It is not clear to me, why the `getting` does merging? 

i'd say getting should return 

```
const compaction_config&  disk_log_impl::compaction_config() const;
```

and the merging should happen elsewhere. I don't see the value of merging in the member function. 

Additionally, you can write a unit test for the merging of the configs to make sure it stays working

Reply:
This is why I haven't named it `get_compaction_config()`. This design is there from the beginning. Log manager calls `compact(compaction_config)`. Log has to apply overrides internally.

Reply:
i like the idea of `merge_cfg_with(defaults)`